protocol/app/prepare/other_msgs.go

```diff
@@ -46,6 +54,18 @@ func RemoveDisallowMsgs(ctx sdk.Context, decoder sdk.TxDecoder, txs [][]byte) []
 
 	var filteredTxs [][]byte
 	for i, txBytes := range txs {
+		// If tx is index 0 and VE enabled, this tx is a VE tx and should be decoded with the extCommitCodec.
+		if i == constants.OracleInfoIndex && ve.VoteExtensionsEnabled(ctx) {
+			_, err := extCommitCodec.Decode(txBytes)
```

Comment:
This feels like a waste of work (ie decoding). In the final state, the protocol will expect the first txbytes to be VE related. 

Can we file a ticket to remove/skip this? I think we should either pass in `txs[1:]` as an input after the upgrade.

Reply:
yup working towards that now

Reply:
take a look a neweest implementation